dev/ci/.gitlab-ci-ctests.yml

```
@@ -0,0 +1,51 @@
+# ==========================================================================
+# Templates and Configuration for CTest-Based Testing
+# ==========================================================================
+#
+# This file defines jobs and templates for running CMake/CTest-based fuctional
+# tests in the global-workflow project.  The CTests are designed to
+# test specific Rocoto Jobs singluarly with predefined input data.
+#
+# CTests are primarily triggered via GitHub API for quick validation of PRs,
+# whereas the full experiment cases are used for more extensive testing.
+#
+# Key components:
+# - .create_ctests: Template for setting up the CMake/CTest environment
+# - .run_ctests_template: Template for executing specific CTest labels
+#
+# CTests are defined in the dev/ctests directory and are categorized by labels,
+# which are referenced in the hosts file to determine which tests to run on
+# which platforms.
+
+# =======================================
+# Templates for CTests
+# =======================================
+
+# Setup job for ctests using CMake
+.create_ctests:
+  extends: .base_config
+  stage: create_experiments
+  script:
+    - echo "Setting up for ctests workflow"
+    - source ${HOMEGFS}/dev/ci/platforms/config.${machine}
+    - cd ${HOMEGFS}/dev/ctests
+    - mkdir -p build
+    - cd build
+    - cmake ../../..
```

```suggestion
    - cmake -S ${HOMEGFS}
```
Specify the location of the source directory to avoid relative paths